bodo/io/iceberg_parquet_reader.cpp

```diff
@@ -673,6 +673,7 @@ PyObject* IcebergParquetReader::get_dataset() {
         throw std::runtime_error("python");
     }
 
+    Py_DECREF(this->catalog);
```

Comment:
A comment clarifying that IcebergParquetReader owns the `catalog` reference and needs to clean it up is useful.